components/director/internal/tenantmapping/model.go

```diff
@@ -126,6 +128,34 @@ func (d *ReqData) GetScopes() (string, error) {
 	return "", apperrors.NewKeyDoesNotExistError(ScopesKey)
 }
 
+// GetGroups returns group name or empty string if there's no group
+func (d *ReqData) GetGroups() ([]string, error) {
+	if groupsVal, ok := d.Body.Extra[GroupsKey]; ok {
+
+		groupsArray, arrayConvertedOk := groupsVal.([]interface{})
```

Comment:
Not sure, but maybe you can try to convert to []string directly?

Reply:
It doesn't allow to convert directly to []string.

Reply:
Ok, I see. How about this implementation?

```go
	userGroups := []string{}
	groupsVal, ok := d.Body.Extra[GroupsKey]
	if !ok {
		return userGroups
	}

	if groupsArray, ok := groupsVal.([]interface{}); ok {
		for _, group := range groupsArray {
			groupString, err := str.Cast(group)
			if err != nil {
				log.Infof("%+v skipped because string conversion failed", group)
				continue
			}

			userGroups = append(userGroups, groupString)
		}
	}

	return userGroups
```